plutus-contract/src/Plutus/Contract/StateMachine/OnChain.hs

```diff
@@ -65,18 +64,23 @@ data StateMachine s i = StateMachine {
 
       -- | The 'AssetClass' of the thread token that identifies the contract
       --   instance.
-      smThreadToken :: Maybe AssetClass
+      smThreadToken :: Maybe TT.ThreadToken
     }
 
-{-# INLINABLE threadTokenValue #-}
+{-# INLINABLE threadTokenValueInner #-}
 -- | The 'Value' containing exactly the thread token, if one has been specified.
-threadTokenValue :: StateMachine s i -> Value
-threadTokenValue StateMachine{smThreadToken} = maybe mempty (\c -> Value.assetClassValue c 1) smThreadToken
+threadTokenValueInner :: Maybe TT.ThreadToken -> ValidatorHash -> Value
+threadTokenValueInner = maybe (const mempty) (TT.threadTokenValue . TT.ttCurrencySymbol)
```

Comment:
am I the only person who likes pattern matching any more? :p

Reply:
😆 
I wanted to write it as `foldMap (TT.threadTokenValue . TT.ttCurrencySymbol)` but the Plutus Monoid class doesn't have an instance for functions.